lms/djangoapps/instructor/views/api.py

```diff
@@ -1031,6 +1031,36 @@ def get_students_features(request, course_id, csv=False):  # pylint: disable=red
 
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def get_students_who_may_enroll(request, course_id):
+    """
+    Initiate generation of a CSV file containing information about
+    students who may enroll in a course.
+
+    Responds with JSON
+        {"status": "... status message ..."}
+
+    """
+    course_key = CourseKey.from_string(course_id)
+    query_features = ['email']
+    try:
+        instructor_task.api.submit_calculate_may_enroll_csv(request, course_key, query_features)
+        success_status = _(
+            "Your students who may enroll report is being generated! "
+            "You can view the status of the generation task in the 'Pending Instructor Tasks' section."
+        )
+        return JsonResponse({"status": success_status})
+    except AlreadyRunningError:
+        already_running_status = _(
+            "A students who may enroll report generation task is already in progress. "
+            "Check the 'Pending Instructor Tasks' table for the status of the task. "
+            "When completed, the report will be available for download in the table below."
```

Comment:
@mhoeber This is a breaking change.

It affects the following test: [`test_calculate_report_csv_already_running`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/tests/test_api.py#L2289)

The test breaks because it requires "already running" messages for CSV generation tasks to conform to the following format:

    {report_type} report generation task is already in progress. 
    Check the 'Pending Instructor Tasks' table for the status of the task. 
    When completed, the report will be available for download in the table below.

Currently, all of the CSV generation tasks produce "already running" messages that conform to this format:

* Grade reports:

        A grade report generation task is already in progress. 
        Check the 'Pending Instructor Tasks' table for the status of the task. 
        When completed, the report will be available for download in the table below.

* Profile information for enrolled students:

        An enrolled student profile report generation task is already in progress.
        Check the 'Pending Instructor Tasks' table for the status of the task.
        When completed, the report will be available for download in the table below.

* List of students who may enroll:

        A students who may enroll report generation task is already in progress. 
        Check the 'Pending Instructor Tasks' table for the status of the task. 
        When completed, the report will be available for download in the table below.

We could of course relax the test to make sure it still passes after applying the changes you requested. However, I think it would be preferable to keep enforcing consistency between "already running" messages of different tasks. One way of achieving this would be to change the "already running" messages for *all* of the CSV generation tasks.

CC @sarina